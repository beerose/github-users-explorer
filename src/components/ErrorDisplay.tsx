/** @jsx jsx */
import { css, jsx } from "@emotion/core";
import React from "react";

import { ErrorMessage } from "../types";

type ErrorDisplayProps = {
  error: ErrorMessage;
};
export const ErrorDisplay: React.FC<ErrorDisplayProps> = ({ error }) => {
  return (
    <section
      css={css`
        display: flex;
        justify-content: center;
        text-align: center;
      `}
    >
      {error.message}
    </section>
  );
};
