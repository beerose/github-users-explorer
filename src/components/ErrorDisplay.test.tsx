import React from "react";
import "@testing-library/jest-dom/extend-expect";

import { render } from "../../test/testUtils";
import { ErrorDisplay } from "./ErrorDisplay";

describe("ErrorDisplay", () => {
  const error = { message: "I'm an error message" };

  it("displays proper error message", () => {
    const { getByText } = render(<ErrorDisplay error={error} />);

    expect(getByText(error.message)).toBeInTheDocument();
  });

  it("matches snapshots", () => {
    const { asFragment } = render(<ErrorDisplay error={error} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
