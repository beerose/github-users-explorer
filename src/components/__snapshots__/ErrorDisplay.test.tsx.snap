// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorDisplay matches snapshots 1`] = `
<DocumentFragment>
  <section
    class="css-rt7554"
  >
    I'm an error message
  </section>
</DocumentFragment>
`;
